### PR TITLE
Fixes #7181

### LINKED ISSUE #7181: EmbedMultipleConfs coordMap problem
**Describe the bug**

When I use the coordMap option of EmbedMultipleConfs to generate small molecule conformations, the generated small molecules are not completely aligned together as expected, but produce two clusters of conformations similar to mirror images.

**To Reproduce**
```from rdkit import Chem
from rdkit.Chem import AllChem, rdMolDescriptors
from rdkit.Chem.AllChem import AlignMol

reference_mol = Chem.SDMolSupplier("ligand.sdf")[0]
mcs_mol = Chem.MolFromSmiles("NC(NC1=CC=C(OC(F)(F)F)C=C1)=O")
cry_match = reference_mol.GetSubstructMatch(mcs_mol)
name = "alignment"
smiles = "CN(C1CCN(C(=O)Nc2ccc(OC(F)(F)F)cc2)CC1)S(C)(=O)=O"

def align(name,smiles):
    simi_mol  = Chem.MolFromSmiles(smiles)
    candi_matches = simi_mol.GetSubstructMatches(mcs_mol)
    for match_idx,candi_match in enumerate(candi_matches):
        match = list(zip(candi_match, cry_match))
        print(match)
        cmap = {pi: reference_mol.GetConformer().GetAtomPosition(xi) 
            for pi, xi in match}
        mol = Chem.Mol(simi_mol)
        mol = Chem.AddHs(mol)
        code = AllChem.EmbedMultipleConfs(mol, coordMap=cmap, numConfs=100)
        keep=[]
        for index, t in enumerate(code):
            conf = mol.GetConformer(t)
            new_mol = Chem.Mol(mol)
            new_mol.RemoveAllConformers()
            new_mol.AddConformer(Chem.Conformer(conf))
            rms = AlignMol(new_mol, reference_mol, atomMap=match)
            # print(index,rms)
            keep.append(new_mol)
        w1 = Chem.SDWriter(f"{name}.sdf")
        for m in keep:
            w1.write(m)
        w1.close()
align(name,smiles)
```

**Screenshots**
image
---
input file:[ligand.sdf.txt](https://github.com/rdkit/rdkit/files/14371358/ligand.sdf.txt)
output file:[alignment.sdf.txt](https://github.com/rdkit/rdkit/files/14371356/alignment.sdf.txt)

### PR DESCRIPTION
**Pull request by greglandrum**
The problem was that ET terms were being applied within the atoms that were in the coordmap. This disables that and adds tighter distance constraints between coordMap atoms when doing the ETK minimization.

### CODE REVIEW

**Comment by greglandrum** [comment]
@sriniker do you have time to take a look at this?

File: Code/DistGeom/DistGeomUtils.cpp [hunk 1]
```diff
@@ -347,13 +348,22 @@ ForceFields::ForceField *construct3DForceField(
...
+        fdist = etkdgDetails.boundsMatForceScaling * 10.0;
         double l = mmat.getLowerBound(i, j);
         double u = mmat.getUpperBound(i, j);
+        if (etkdgDetails.constrainedAtoms.size() &&
+            etkdgDetails.constrainedAtoms[i] &&
+            etkdgDetails.constrainedAtoms[j]) {
+          // we're constrained, so use very tight bounds
+          l = u = ((*positions[i]) - (*positions[j])).length();
+          l -= 0.01;
+          u += 0.01;
```
Comment (ptosco) [inline Code/DistGeom/DistGeomUtils.cpp:364]:
```suggestion
          constexpr double INCR = 0.01;
          l -= INCR;
          u += INCR;
```

File: Code/DistGeom/DistGeomUtils.cpp [hunk 2]
```diff
@@ -347,13 +348,22 @@ ForceFields::ForceField *construct3DForceField(
...
-  // minimum distance for all other atom pairs
-  fdist = etkdgDetails.boundsMatForceScaling * 10.0;
+  // minimum distance for all other atom pairs that aren't constrained
   for (unsigned int i = 1; i < N; ++i) {
     for (unsigned int j = 0; j < i; ++j) {
       if (!atomPairs[j * N + i]) {
+        fdist = etkdgDetails.boundsMatForceScaling * 10.0;
         double l = mmat.getLowerBound(i, j);
         double u = mmat.getUpperBound(i, j);
+        if (etkdgDetails.constrainedAtoms.size() &&
```
Comment (ptosco) [inline Code/DistGeom/DistGeomUtils.cpp:358]:
```suggestion
        if (!etkdgDetails.constrainedAtoms.empty() &&
```
I find the above clearer. Feel free to ignore the suggestion.

**Review by ptosco — APPROVED** [review]
LGTM. I added two suggestions that you may ignore.

**Pull request merged**